### PR TITLE
fix(49426): Object method snippet completions incorrectly add this parameters

### LINKED ISSUE #49426: Object method snippet completions incorrectly add this parameters
# Bug Report

### 🔎 Search Terms
- suggest / suggestions
- completions
- IntelliSense

### 🕗 Version & Regression Information
4.8.0-dev.20220606, not a regression

### 💻 Code

Try using a method snippet that uses a `this` parameter 

```ts
interface IFoo {
    bar(this: IFoo): void; 
}

const foo: IFoo = {
    // Trigger completions here    
};

foo.bar();
```

### 🙁 Actual behavior

The `this` parameter is also inserted:

```ts
interface IFoo {
    bar(this: IFoo): void;
}

const foo: IFoo = {
    bar(this) {

    },
};

foo.bar();
```

### 🙂 Expected behavior

The this parameter should likely be omitted in this case

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #49426

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -4664,6 +4664,7 @@ namespace ts {
         UseAliasDefinedOutsideCurrentScope      = 1 << 14,  // Allow non-visible aliases
         UseSingleQuotesForStringLiteralType     = 1 << 28,  // Use single quotes for string literal type
         NoTypeReduction                         = 1 << 29,  // Don't call getReducedType
+        OmitThisParameter                       = 1 << 30,
```
Comment (gabritto) [inline src/compiler/types.ts:4667]: For consistency, I think this flag should also be added to `TypeFormatFlags`, and then maybe look at `signatureToString`, `typeToString` etc to see if they need updating to be able to handle the new flag by omitting `this` params.

**Review by gabritto — CHANGES_REQUESTED** [review]
Left a comment on adding an equivalent type flag, but the rest looks good. I'll also try to get someone else to review it to make sure we're ok with adding a new `NodeBuilderFlag`.

File: src/compiler/types.ts (hunk 1 above)
Comment (a-tarasyuk) [reply src/compiler/types.ts:4667]: @gabritto Thanks for the review. I've added `OmitThisParameter` to `TypeFormatFlags`

File: src/compiler/types.ts [hunk 2]
```diff
@@ -4664,10 +4664,11 @@ namespace ts {
...
```
Comment (gabritto) [inline src/compiler/types.ts:4667]: @weswigham @rbuckton could one of you take a look at this and see if we're ok with adding this flag? I'm also not sure we can use `1 << 30`. `1 << 25` seems to be free though, but again, I'm not sure if we're ok with adding a flag here.

**Review by weswigham — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Pull request merged**